data/settings/1.15.x/boot.toml

```diff
@@ -0,0 +1,88 @@
+[[docs.ref.init-parameters]]
+description = """
+Init parameters expressed as key/value pairs. 
+If boot data exists at `/proc/bootconfig`, Bottlerocket generates the settings from this data on first boot.
+During the boot process, the init parameters pass via the kernel command line.
+"""
+see = [
+    ["[Boot Configuration on kernel.org](https://www.kernel.org/doc/html/latest/admin-guide/bootconfig.html)"]
+]
+[[docs.ref.init-parameters.example]]
+direct_toml = """
+[settings.boot.init-parameters]
+"log_level" = ["debug"]
+"splash" = []
+"""
+direct_shell = """
+apiclient apply <<EOF
+[settings.boot.init-parameters]
+"log_level" = ["debug"]
+"splash" = []
+EOF
+"""
+
+
+
+[[docs.ref.kernel-parameters]]
+description = """
+Kernel parameters expressed as key/value pairs. 
+If boot data exists at `/proc/bootconfig`, Bottlerocket generates the settings from this data on first boot.
+During the boot process, the init parameters pass via the kernel command line.
+"""
+see = [
+    ["[Boot Configuration on kernel.org](https://www.kernel.org/doc/html/latest/admin-guide/bootconfig.html)"]
+]
+[[docs.ref.kernel-parameters.example]]
+direct_toml = """
+[settings.boot.kernel-parameters]
+"console" = [
+  "tty0",
+  "ttyS1,115200n8",
+]
+"crashkernel" = [
+  "2G-:256M"
+]
+"slub_debug" = [
+  "options,slabs"
+]
+"usbcore.quirks" = [
+  "0781:5580:bk",
+  "0a5c:5834:gij"
+]
+"""
+direct_shell = """
+apiclient apply <<EOF
+[settings.boot.kernel-parameters]
+"console" = [
+  "tty0",
+  "ttyS1,115200n8",
+]
+"crashkernel" = [
+  "2G-:256M"
+]
+"slub_debug" = [
+  "options,slabs"
+]
+"usbcore.quirks" = [
+  "0781:5580:bk",
+  "0a5c:5834:gij",
+]
+EOF
+"""
+
+[[docs.ref.reboot-to-reconcile]]
+description = """
+When `true`, anytime the values for `settings.boot.kernel-parameters` or `settings.boot.init-parameters` change, the node reboots.
```

Comment:
The reboot will only occur *during boot*, not anytime `kernel-parameters` or `init-parameters` changes. Towards the end of a host's initial boot, prairiedog may notice the current `/proc/bootconfig` (the bootconfig the kernel booted with) does not match what has meanwhile been configured on the host (e.g. via user data or a bootstrap container) and has been persisted as the bootconfig to be used during the next boot. It will then trigger a reboot to reconcile those two. On the next boot, they will match.

This really is about first boot behavior: The kernel is already running by the time Bottlerocket can read any user data. By then it is too late to change some of the kernel's (or init's/systemd's) settings, so the only way for those changes to take effect is to reboot. Changing `kernel-parameters` or `init-parameters` during runtime will not trigger a reboot, but only persist the new bootconfig to be used during next boot.

Reply:
Thanks! I think I've got it.